packages/types/src/srml/parachains/types.ts

```diff
@@ -0,0 +1,167 @@
+// Auto-generated, do not edit
+
+import { Enum, Struct, Tuple, Vector } from '../../codec';
+import { AccountId, Bytes, Hash, Signature, U32, u64 } from '../../primitive';
+
+export interface AttestedCandidate extends Struct {
+  readonly candidate: CandidateReceipt;
+  readonly validityVotes: Vector<ValidityVote>;
+  readonly availabilityVotes: Vector<AvailabilityVote>;
+}
+
+export interface AuctionIndex extends U32 {}
+
+export interface AvailabilityVote extends Tuple {}
+
+export interface BalanceUpload extends Tuple {}
+
+export interface Bidder extends Enum {
+  /**
+   * @description 0:: New(NewBidder)
+   */
+  readonly isNew: boolean;
+  readonly asNew: NewBidder;
+  /**
+   * @description 1:: Existing(ParaId)
+   */
+  readonly isExisting: boolean;
+  readonly asExisting: ParaId;
+}
+
+export interface CandidateReceipt extends Struct {
+  readonly parachainIndex: ParaId;
+  readonly collator: AccountId;
+  readonly signature: CollatorSignature;
+  readonly headData: HeadData;
+  readonly balanceUploads: Vector<BalanceUpload>;
+  readonly egressQueueRoots: Vector<EgressQueueRoot>;
+  readonly fees: u64;
+  readonly blockDataHash: Hash;
+}
+
+export interface CollatorSignature extends Signature {}
+
+export interface EgressQueueRoot extends Tuple {}
```

Comment:
These should really convey more information, i.e. `EgressQueueRoot = (ParaId, Hash)` (as defined), here none of the actual part info is available.

This is not valid (`An interface can only extend an identifier/qualified-name with optional type arguments.ts(2499)`) -

```js
export interface EgressQueueRoot extends [ParaId, Hash] implements Codec {}
```

However this work-around does do the trick -

```js
export type EgressQueueRootTuple = [ParaId, Hash];

export interface EgressQueueRoot extends EgressQueueRootTuple implements Codec {}
```

i.e. in the above you can do `[paraId, hash] = foo` (where foo is `EgressQueueRoot`) and `a` & `b` would have the `ParaId` and `Hash` types respectively.